### ISSUE TITLE
Mac client intermittently crashes silently with no errors

### ISSUE THREAD

**Issue by nbolton** [symless/synergy-core#4691]
1. Start Windows server with SSL
2. Connect Mac client with SSL
3. Stop and start Windows server repeatedly
4. Mac client stops without any errors (low probability)
5. Mac client 'Start' button is enabled

![screen shot 2015-05-19 at 2 53 14 pm](https://cloud.githubusercontent.com/assets/313427/7704530/ee5b5d74-fe36-11e4-8dc0-d1bae04779d4.png)

**Comment by moltendorf** [comment]
I have a (fairly) reliable way to reproduce a Mac client crash on my setup:

Mac Model: https://support.apple.com/kb/SP710?locale=en_US (2.6GHz base configuration; OS X 10.10).
Energy Saver Settings: https://moltendorf.net/share/Pictures/Random/Energy-Saver-Settings.png
Synergy Version: Synergy Pro 1.7.2.

1. Wait for Mac to turn off display from being idle (I also tried with 1 minute instead of 3 hours: crash).
2. Use mouse on server computer to wake monitor by moving it to Mac's monitor.
3. Mac's screen wakes up.
4. Synergy crashes and does not restart.

If I replace Step 2 with "use my trackpad to wake my Mac", then Synergy does not crash.
It's an Apple Magic Trackpad which is connected via Bluetooth to the Mac Mini. (I noticed this because Synergy would crash as mentioned when I would be gone for a while, so I just went straight to using the trackpad to wake, as I expected to have to use it to start Synergy back up, only Synergy would not crash.)

I thought Synergy was crashing more often at first, but I haven't noticed it doing that for some reason: only this specific pattern (maybe it was 1.7.1 that I remember it crashing more often).

As another side note, powerd process on my Mac goes absolutely haywire after being up for 24 hours so I generally kill it because it uses 80%+ CPU (I haven't been able to figure out how to fix it, and it doesn't seem to cause any issues not running), it may related.

**Comment by nbolton** [comment]
@moltendorf Do you think this might be a duplicate of #4558? The difference seems to be in this case, no error is raised on exit.

**Comment by ylluminate** [comment]
Having a crash here as well, not sure if related.  I see a couple of Mac crashes floating around the issues and so I'm going to insert this link for review on this thread and then see if you would like for me to create a new issues @nbolton: https://gist.github.com/323489987744959a1cfd

Presently I was setting it up on a new OS X 10.10.4 beta workstation and nearly immediately upon startup (as a client) I receive this.

**Comment by nbolton** [comment]
@ylluminate Do you see a silent crash? i.e. Nothing in the log indicates errors...

**Comment by moltendorf** [comment]
Still seeing the crash in 1.7.3.

New log ending: 
    WARNING: cursor may not be visible
    INFO: entering screen
    E
    R
    ROR: ssl error
    NOTE: stopping synergy desktop process

**Comment by nbolton** [comment]
@moltendorf We moved it to 1.7.4

**Comment by XinyuHou** [comment]
@moltendorf  @ylluminate 
Please test this nightly.

http://synergy-project.org/nightly?filter=121080

If the issue persists, please send us both server and client log information.

**Comment by speaker** [comment]
@moltendorf @ylluminate 
I've tried reproducing this behavior with the newest code without any success. Would you please try http://synergy-project.org/nightly?filter=v1.7.4-stable-267f3ac and see if this is still happening?

thank you!

**Comment by moltendorf** [comment]
I've installed them and there doesn't seem to be any stability issues so far. I've repeated my test that always crashed it in 1.7.2 five times without failure; so I'd say my description of this problem is fixed. In 1.7.3 I had a lot of other issues (they had tickets made by other people, that I didn't feel the need to comment on) that looked far worse than this one so I didn't really notice this issue behind all of the other noise. I'll see how long this stability lasts as I leave my computers on, reboot them, and more over time. Thanks so much for fixing this.

**Comment by speaker** [comment]
Good to hear. It's difficult to know which of the fixes addressed this specifically, but there were plenty of candidates.

Thank you for the feedback.

**Comment by nbolton** [comment]
Still happening for me with 88f27e6

**Comment by XinyuHou** [comment]
Please test this nightly 79bb437
http://synergy-project.org/nightly?filter=79bb437

If you use SSL and have trouble with downloading plugins
1. Stop synergy first
2. Delete the old plugin directory
3. Rerun wizard to download again

Plugin directory:
Windows: %User directory%\AppData\Local\Synergy\Plugins
Mac: ~/Library/Synergy/Plugins
Linux: ~/.synergy/plugins

**Comment by byannoni** [comment]
I encounter this issue on Arch Linux (Linux 4.0.6) running 79bb437 with a Windows 7 Pro x64 server which is running 1.7.3.

**Comment by XinyuHou** [comment]
@byannoni 
Do you use SSL?

**Comment by byannoni** [comment]
No, I do not use SSL.

**Comment by XinyuHou** [comment]
@byannoni 
Please test 79bb437 on both your server and client. If issue persists, please send us some log.

**Comment by byannoni** [comment]
The issue occurred again my Arch Linux x64 client and Windows 7 x64 server. Both were running 79bb437, no SSL.

Here is the server log:
```
[2015-07-25T15:59:27] NOTE: client "archlinux-desktop" has connected
[2015-07-25T15:59:28] INFO: switch from "FastHP1" to "archlinux-desktop" at 1439,210
[2015-07-25T15:59:28] INFO: leaving screen
[2015-07-25T15:59:28] DEBUG: open clipboard
[2015-07-25T15:59:28] DEBUG: close clipboard
[2015-07-25T15:59:28] INFO: screen "FastHP1" updated clipboard 0
[2015-07-25T15:59:28] DEBUG: open clipboard
[2015-07-25T15:59:28] DEBUG: close clipboard
[2015-07-25T15:59:28] INFO: screen "FastHP1" updated clipboard 1
[2015-07-25T16:04:26] DEBUG: received client "archlinux-desktop" grabbed clipboard 0 seqnum=9
[2015-07-25T16:04:26] INFO: screen "archlinux-desktop" grabbed clipboard 0 from "FastHP1"
[2015-07-25T16:04:26] DEBUG: open clipboard
[2015-07-25T16:04:26] DEBUG: empty clipboard
[2015-07-25T16:04:26] DEBUG: close clipboard
[2015-07-25T16:04:26] DEBUG: clipboard changed: synergy owned
[2015-07-25T16:04:43] INFO: switch from "archlinux-desktop" to "FastHP1" at 7,317
[2015-07-25T16:04:43] INFO: entering screen
[2015-07-25T16:04:43] DEBUG: open clipboard
[2015-07-25T16:04:43] DEBUG: empty clipboard
[2015-07-25T16:04:43] DEBUG: close clipboard
[2015-07-25T16:04:43] NOTE: client "archlinux-desktop" has disconnected
[2015-07-25T16:04:43] DEBUG: active sides: 2
[2015-07-25T16:04:48] DEBUG: clipboard changed: lost ownership
[2015-07-25T16:04:48] INFO: screen "FastHP1" grabbed clipboard 0 from "archlinux-desktop"
[2015-07-25T16:04:48] INFO: screen "FastHP1" grabbed clipboard 1 from "FastHP1" "archlinux-desktop"
```

And the client:
```
NOTE: started client
NOTE: connecting to '10.255.253.106': 10.255.253.106:24800
DEBUG: event queue is ready
DEBUG: open clipboard 0
DEBUG: empty clipboard 0
DEBUG: grabbed clipboard 0
DEBUG: close clipboard 0
DEBUG: open clipboard 1
DEBUG: empty clipboard 1
DEBUG: grabbed clipboard 1
DEBUG: close clipboard 1
NOTE: connected to server
INFO: entering screen
DEBUG: send xscreensaver command: 311 0 0
DEBUG: lost clipboard 0 ownership at time 1293889195
DEBUG: lost clipboard 0 ownership at 1293889195
INFO: leaving screen
DEBUG: open clipboard 0
DEBUG: ICCCM fill clipboard 0
DEBUG:   available targets: TIMESTAMP (405), TARGETS (404), MULTIPLE (401), _VIMENC_TEXT (665), _VIM_TEXT (664)
DEBUG:   added format 0 for target UTF8_STRING (324) (4355 bytes)
DEBUG: close clipboard 0
```

**Comment by XinyuHou** [comment]
@byannoni 
Did you use 79bb437 on client side as well?

**Comment by byannoni** [comment]
Yes, both were running 79bb437.

**Comment by XinyuHou** [comment]
@byannoni 
It seems you are still using an old version of Synergy on client, since there is no time stamp in your logging.

**Comment by byannoni** [comment]
Yes, I realized that after the fact. I have recompiled with 79bb437 and am using that now. Sorry for the false alarm.

**Comment by nbolton** [comment]
Just saw this again after locking my screen using shift+fn+eject.

![screen shot 2015-07-27 at 4 06 02 pm](https://cloud.githubusercontent.com/assets/313427/8920031/896a540a-3479-11e5-997d-d0c9ec70ec47.png)

**Comment by nbolton** [comment]
@byannoni Please open a new issue, your issue is not related (this is Mac only).

**Comment by nbolton** [comment]
Sorry this still isn't fixed, enhancement #4901 should help in the meantime.

**Comment by nbolton** [comment]
Some developments, looks like #4901 helped. I just saw this log message...

```
[2015-08-06T12:39:22] DEBUG: clipboard changed
[2015-08-06T12:39:47] INFO: leaving screen
[2015-08-06T12:39:47] DEBUG: hiding cursor
[2015-08-06T12:39:47] DEBUG: clipboard changed
[2015-08-06T12:39:47] DEBUG: opening clipboard
[2015-08-06T12:39:47] DEBUG: opening clipboard
[2015-08-06T12:39:47] WARNING: cursor may not be visible
[2015-08-06T12:39:49] INFO: process exited normally
[2015-08-06T12:39:49] INFO: detected process not running, auto restarting
[2015-08-06T12:39:50] DEBUG: starting process
[2015-08-06T12:39:50] INFO: starting client
```

Weirdly, the process just exited with code 0 and no errors. Seemed to be related to the clipboard? It repeated the "opening clipboard" message twice before exiting.
